crates/nargo/src/ops/test.rs

```diff
@@ -0,0 +1,68 @@
+use acvm::{acir::native_types::WitnessMap, Backend};
+use noirc_driver::{compile_no_check, CompileOptions};
+use noirc_errors::FileDiagnostic;
+use noirc_frontend::hir::{def_map::TestFunction, Context};
+
+use super::execute_circuit;
+
+pub enum TestStatus {
+    Pass,
+    Fail { message: String },
+    CompileError(FileDiagnostic),
+}
+
+pub fn run_test<B: Backend>(
```

Comment:
```suggestion
pub fn run_test<B: BlackBoxFunctionSolver>(
```

We can relax this trait bound.